### PR TITLE
ikev1: added ikev1 test - v3

### PR DESCRIPTION
**Pull request by fhonza**
This PR adds checks for a new IKEv1 parser for suricata, which is currently in review.
It checks for metadata logging and keywords triggering alerts.

### CODE REVIEW

**Review by catenacyber — CHANGES_REQUESTED** [review]
Thanks for this work.

CI is not happy :-(

Could you review your commits so that each one compiles ?

And have them formatted ok with `git rebase master -x "git clang-format HEAD^"` ?

File: tests/ikev2-weak-dh/test.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ requires:
     - HAVE_LIBJANSSON
     - RUST
   min-version: 4.1.0
+  lt-version: 6
```
Comment (catenacyber) [inline tests/ikev2-weak-dh/test.yaml:6]: @jasonish this test is getting deprecated and replaced by `ike-weak-dh`
Is this the way to go ?

@fhonza could you modify ikev2-weak-dh` README (or add a comment in this yaml) to explain this ?

Comment (fhonza) [reply tests/ikev2-weak-dh/test.yaml:6]: with the unification of ikev1 and ikev2 into ike both parsers now use output-json-ike.c, which just adds the event_type "ike" (if i've debugged correctly), which is breaking the old test. Perhaps i can try to use either ikev1 or ikev2 depending on the protocol version.
In this function (JsonIKELogger(...)) I have the state as a parameter, perhaps it's possible to check there.

In this case the existing test could be left untouched, would be better?

Comment (catenacyber) [reply tests/ikev2-weak-dh/test.yaml:6]: I think event_type "ike" is good.
@jasonish what do you think ?

Comment (jasonish) [reply tests/ikev2-weak-dh/test.yaml:6]: Just "ike" would be idea I think.

Comment (catenacyber) [reply tests/ikev2-weak-dh/test.yaml:6]: Does the framework enable to have some checks with some version and some others checks with other versions ?
cc @inashivb 

That is what we want here I think.
The pcap is the same.
Previously, we checked `ikev2` field in eve.json, and now we will want it to be simply `ike`

**Comment by catenacyber** [comment]
@fhonza what are the changes with your force-push ?

**Comment by fhonza** [comment]
> @fhonza what are the changes with your force-push ?

just pushed the wrong branch, was preparing v4 PR, just added a comment on the deprecated test...

but discussion seems to go on, i'll wait a bit with it :)

File: tests/ikev2-weak-dh/test.yaml (hunk 1 above)
Comment (catenacyber) [reply tests/ikev2-weak-dh/test.yaml:6]: Thanks Shivani.
Could you do this @fhonza ?

Comment (fhonza) [reply tests/ikev2-weak-dh/test.yaml:6]: great! created a new PR...

**Pull request closed without merging**